src/uwp/build.cmd

```diff
@@ -0,0 +1,110 @@
+@if not defined _echo @echo off
+setlocal
+
+:SetupArgs
+:: Initialize the args that will be passed to cmake
+set __nativeWindowsDir=%~dp0Windows
+set __binDir=%~dp0..\..\Bin
+set __rootDir=%~dp0..\..
+set __CMakeBinDir=""
+set __IntermediatesDir=""
+set __BuildArch=x64
+set __appContainer=""
+set __VCBuildArch=x86_amd64
+set CMAKE_BUILD_TYPE=Debug
+set "__LinkArgs= "
+set "__LinkLibraries= "
+set __PortableBuild=0
+
+:Arg_Loop
+if [%1] == [] goto :ToolsVersion
+if /i [%1] == [Release]     ( set CMAKE_BUILD_TYPE=RelWithDebInfo&&shift&goto Arg_Loop)
+if /i [%1] == [Debug]       ( set CMAKE_BUILD_TYPE=Debug&&shift&goto Arg_Loop)
+
+if /i [%1] == [AnyCPU]      ( set __BuildArch=x64&&set __VCBuildArch=x86_amd64&&shift&goto Arg_Loop)
+if /i [%1] == [x86]         ( set __BuildArch=x86&&set __VCBuildArch=x86&&shift&goto Arg_Loop)
+if /i [%1] == [arm]         ( set __BuildArch=arm&&set __VCBuildArch=x86_arm&&set __SDKVersion="-DCMAKE_SYSTEM_VERSION=10.0"&&shift&goto Arg_Loop)
+if /i [%1] == [x64]         ( set __BuildArch=x64&&set __VCBuildArch=x86_amd64&&shift&goto Arg_Loop)
+if /i [%1] == [amd64]       ( set __BuildArch=x64&&set __VCBuildArch=x86_amd64&&shift&goto Arg_Loop)
+
+if /i [%1] == [rid]         ( set __TargetRid=%2&&shift&&shift&goto Arg_Loop)
+shift
+goto :Arg_Loop
+
+:ToolsVersion
+:: Determine the tools version to pass to cmake/msbuild
+if not defined VisualStudioVersion (
+    if defined VS140COMNTOOLS (
+        goto :VS2015
+    ) 
+    goto :MissingVersion
+) 
+if "%VisualStudioVersion%"=="14.0" (
+    goto :VS2015
+) 
+
+:MissingVersion
+:: Can't find VS 2013+
+echo Error: Visual Studio 2015 required  
```

Comment:
We do not support building across the stack with VS 2017 (it changed a lot of dependencies). We can look into that once we do intend to support building using it in (atleast in official builds).